sd/etcd/registrar.go

```diff
@@ -69,15 +69,18 @@ func NewRegistrar(client Client, service Service, logger log.Logger) *Registrar
 
 // Register implements the sd.Registrar interface. Call it when you want your
 // service to be registered in etcd, typically at startup.
-func (r *Registrar) Register() {
+func (r *Registrar) Register() error {
```

Comment:
I don't think it's clear how to use this type/method. What should happen if the first Register fails? Should it still `go r.loop()` or not? If yes, does it really matter if the first register fails? If not, should the loop happen in a different way?